Game.cs

```diff
@@ -1,9 +1,50 @@
 
 class Game
 {
-  public void Run(){
-    Grid grid = new Grid(10, 10);
-    ASCIIGraphics graphics = new ASCIIGraphics();
-    graphics.Draw(grid);
+  public void Run()
+  {
+    TerminalInputOutput input_output = new TerminalInputOutput();
+    LoadSave load_save = new();
+
+    Args args = input_output.ParseInput();
+    Grid grid = CreateGrid(load_save, args);
+    
+    input_output.Draw(grid);
   }
-}
\ No newline at end of file
+
+  private Grid CreateGrid(LoadSave load_save, Args args)
+  {
+    if(args.GameMode == GameMode.LoadMaze)
+    {
+      return load_save.Load(args.Path);
+    }
+
+
+    Algorithm algorithm;
+    if(args.Algorithm == AlgType.SideWinder)
```

Comment:
It is better using `switch-case` for `enum`'s, since it is easier to support additional values if necessary.